haskell_scotty/ClientRepo.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE OverloadedStrings #-}
+module ClientRepo where
+
+import Client
+import DBConfig
+import Database.MySQL.Simple
+import Database.MySQL.Simple.Result
+import Database.MySQL.Simple.QueryResults
+
+all :: IO [Client]
+all = do
+  DBConfig user pass database <- readDBConfig
+  conn <- connect defaultConnectInfo {
+            connectUser = user,
+            connectPassword = pass,
+            connectDatabase = database }
+  clients <- query_ conn "SELECT * FROM clients"
+  return clients
+
+instance QueryResults Client where
+  convertResults [fa,fb,fc,fd,fe,ff,fg,fh,fi,fj,fk,fl] [va,vb,vc,vd,ve,vf,vg,vh,vi,vj,vk,vl] =
+    Client a b c d e f g h i j k l 
+      where a = convert fa va
+            b = convert fb vb
+            c = convert fc vc
+            d = convert fd vd
+            e = convert fe ve
+            f = convert ff vf
+            g = convert fg vg
+            h = convert fh vh
+            i = convert fi vi
+            j = convert fj vj
+            k = convert fk vk
+            l = convert fl vl
```

Comment:
There's a thing called "uncurry" that works on 2-tuples. There is no uncurry13, but you can define it in the obvious way :-)

uncurry13 f (a,b,c,d,e,f,g,h,i,j,k,l,m) = f a b c d e f g h i j k l m

There's some games we can play with `ExistentialTypes`, or like I say there's Template Haskell. I'll get to it at some point.